disnake/automod.py

```diff
@@ -0,0 +1,681 @@
+            You do not have proper permissions to delete the rule.
+        NotFound
+            The rule does not exist.
+        HTTPException
+            Deleting the rule failed.
+        """
+        await self.guild._state.http.delete_auto_moderation_rule(
+            self.guild.id, self.id, reason=reason
+        )
+
+
+class AutoModActionExecution:
+    """Represents the data for an :func:`on_automod_action_execution` event.
+
+    .. versionadded:: 2.6
+
+    Attributes
+    ----------
+    action: :class:`AutoModAction`
+        The action that was executed.
+    guild: :class:`Guild`
+        The guild this action was executed in.
+    rule_id: :class:`int`
+        The ID of the rule that matched.
+    rule_trigger_type: :class:`AutoModTriggerType`
+        The trigger type of the rule that matched.
+    user_id: :class:`int`
+        The ID of the user that triggered this action.
+        See also :attr:`.user`.
+    channel_id: Optional[:class:`int`]
+        The channel or thread ID in which the event occurred, if any.
+        See also :attr:`.channel`.
+    message_id: Optional[:class:`int`]
+        The ID of the message that matched. ``None`` if the message was blocked,
+        or if the content was not part of a message.
+        See also :attr:`.message`.
+    alert_message_id: Optional[:class:`int`]
+        The ID of the alert message sent as a result of this action, if any.
+        See also :attr:`.alert_message`.
+    content: :class:`str`
+        The content that matched.
+
+        Requires :attr:`Intents.message_content` to be enabled,
+        otherwise this field will be empty.
+
+    matched_keyword: Optional[:class:`str`]
+        The keyword that matched.
+    matched_content: Optional[:class:`str`]
+        The substring of :attr:`.content` that matched the rule/keyword.
+
+        Requires :attr:`Intents.message_content` to be enabled,
+        otherwise this field will be empty.
+    """
+
+    __slots__ = (
+        "action",
+        "guild",
+        "rule_id",
+        "rule_trigger_type",
+        "user_id",
+        "channel_id",
+        "message_id",
+        "alert_message_id",
+        "content",
+        "matched_keyword",
+        "matched_content",
+    )
+
+    def __init__(self, *, data: AutoModActionExecutionEventPayload, guild: Guild) -> None:
+        self.guild: Guild = guild
+        self.action: AutoModAction = _automod_action_factory(data["action"])
+        self.rule_id: int = int(data["rule_id"])
+        self.rule_trigger_type: AutoModTriggerType = try_enum(
+            AutoModTriggerType, data["rule_trigger_type"]
+        )
+        self.user_id: int = int(data["user_id"])
+        self.channel_id: Optional[int] = _get_as_snowflake(data, "channel_id")
+        self.message_id: Optional[int] = _get_as_snowflake(data, "message_id")
+        self.alert_message_id: Optional[int] = _get_as_snowflake(data, "alert_system_message_id")
+        self.content: str = data["content"]
```

Comment:
This should not assume that this key actually exists, unfornately.

Reply:
Good catch, even though the API docs say that this field always exists, it'll be missing (instead of `""`) if the message content intent is disabled :/
https://github.com/DisnakeDev/disnake/pull/530/commits/9b8d2f4fce92b2cd0cf34e21be2f0755ba40341d

Reply:
Y'know, I'll make an issue about this.